### DISCUSSION TITLE
[Feature] useMutationObserver

### DISCUSSION THREAD

**Discussion by Dchole** [juliencrn/usehooks-ts#417] [Ideas]
I have a [`useMutationObserver`](https://codesandbox.io/s/mutation-observer-hook-rrzwsw?file=/src/useMutationObserver.ts) hook that handles the [MutationObserver API](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) in my react code.

The MutationObserver API is very similar to the [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). It provides the ability to listen to changes made in the DOM tree. I'll love to make a PR and have it included in the library

**Comment by juliencrn** [comment]
Awesome hook! Yes, it's more than welcome 🙏
You could use the `npm run plop` script to create the hook boilerplate.
Thanks!
